plugins/esi/fetcher/HttpDataFetcherImpl.cc

```diff
@@ -47,10 +47,29 @@ HttpDataFetcherImpl::_release(RequestData &req_data)
 }
 
 HttpDataFetcherImpl::HttpDataFetcherImpl(TSCont contp, sockaddr const *client_addr, const char *debug_tag)
-  : _contp(contp), _n_pending_requests(0), _curr_event_id_base(FETCH_EVENT_ID_BASE), _headers_str(""), _client_addr(client_addr)
+  : _contp(contp), _n_pending_requests(0), _curr_event_id_base(FETCH_EVENT_ID_BASE), _headers_str("")
 {
   _http_parser = TSHttpParserCreate();
   snprintf(_debug_tag, sizeof(_debug_tag), "%s", debug_tag);
+  // default client address to use for fetch url
+  struct sockaddr_in sin;
+  memset(&sin, 0, sizeof(sin));
+  sin.sin_family      = AF_INET;
+  sin.sin_addr.s_addr = inet_addr("127.0.0.1");
```

Comment:
If large number of these objects are constructed it would be worth it to avoid redundant calls to inet_addr():
```
static const auto default_sin_s_addr = inet_addr("127.0.0.1");
sin.sin_addr.s_addr = default_sin_s_addr;
```

Reply:
Actually, he should use `INADDR_LOOPBACK`.

Reply:
Topper.